api/v1beta1/machine_types.go

```diff
@@ -102,6 +110,8 @@ type MachineSpec struct {
 	// Defaults to 10 seconds.
 	// +optional
 	NodeDeletionTimeout *metav1.Duration `json:"nodeDeletionTimeout,omitempty"`
+
+	NodeLabels *map[string]string `json:"nodeLabels,omitempty"`
```

Comment:
The more that I think about the option to add a new field and related implications on the UX up to ClusterClass, the more that I get to the idea that without this field the UX will be cleaner and simple, with a single API field to control plus propagation rules ensuring the labels and annotations) will be consistent across the entire chain, with the only exception of prefix ruling the machine --> node propagation for security reasons.

The key point is how to achieve this by avoiding rollbacks when transitioning... thinking about it